### PR TITLE
Make the regular changes for 4.0.0

### PR DESCRIPTION
**Pull request by timyates**
For some reason, there are 2 issues

- The test-kotlin project just fails, and I cannot see what it is missing
- It seems that activeSpan() always returns null, so the tags are not being added in the tests

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -56,6 +56,8 @@ awssdk-core = { module = 'software.amazon.awssdk:sdk-core' }
 
 micronaut-aws-bom = { module = 'io.micronaut.aws:micronaut-aws-bom' }
 
+javax-annotation-api = { module = "javax.annotation:javax.annotation-api" }
```
Comment (graemerocher) [inline gradle/libs.versions.toml:59]: this shouldn't be used

Comment (timyates) [reply gradle/libs.versions.toml:59]: Any idea how to get around grpc annotating classes it creates with `@javax.annotation.Generated`?

Comment (donbeave) [reply gradle/libs.versions.toml:59]: Seems no way until this issue will be resolved: https://github.com/grpc/grpc-java/issues/9179

**Comment by timyates** [comment]
Superseded by https://github.com/micronaut-projects/micronaut-tracing/pull/213

**Pull request closed without merging**